GetIntoTeachingApi/Utils/CommaSeparatedQueryStringConvention.cs

```diff
@@ -12,9 +13,23 @@ public void Apply(ActionModel action)
             {
                 if (parameter.Attributes.OfType<CommaSeparatedAttribute>().Any() && !parameter.Action.Filters.OfType<CommaSeparatedQueryStringAttribute>().Any())
                 {
-                    parameter.Action.Filters.Add(new CommaSeparatedQueryStringAttribute(parameter.ParameterName, ","));
+                    var attributeNames = parameter.Attributes.OfType<CommaSeparatedAttribute>().First().AttributeNames;
+
+                    if (attributeNames != null)
+                    {
+                        AddToActionFilterAsCommaSeparated(parameter, attributeNames);
+                    }
+                    else
+                    {
+                        AddToActionFilterAsCommaSeparated(parameter, new string[] { parameter.ParameterName });
+                    }
                 }
             }
-        }
+        }
+
+        private static void AddToActionFilterAsCommaSeparated(ParameterModel parameter, string[] keys)
+        {
+            parameter.Action.Filters.Add(new CommaSeparatedQueryStringAttribute(keys, ","));

```

Comment:
Instead of bubbling `keys` all the way down to `CommaSeparatedQueryStringAttribute` would it be clearer/fewer changes to call it multiple times? something like:

```
var attributeNames = parameter.Attributes.OfType<CommaSeparatedAttribute>().First().AttributeNames;
if (attributeNames == null) {
  attributeNames = new string[] { parameter.ParameterName };
}

attributeNames.ForEach(a => parameter.Action.Filters.Add(new CommpaSeparatedQueryStringAttribute(a, ",");
```

Reply:
The `CommaSeparatedQueryStringValueProvider` gets called once per request with a collection of the query parameters so I don't think this will work, it needs access to a list of allowed comma separated keys to check against.